src/sql-parser/src/parser.rs

```diff
@@ -1667,13 +1667,41 @@ impl<'a> Parser<'a> {
                 // FROM S3 BUCKET '<bucket>' OBJECTS FROM SCAN MATCHING '<pattern>'
                 self.expect_keyword(BUCKET)?;
                 let bucket = self.parse_literal_string()?;
-                self.expect_keywords(&[OBJECTS, FROM, SCAN])?;
+                self.expect_keywords(&[OBJECTS, FROM])?;
+                let mut key_sources = Vec::new();
+                let mut parsed_scan = false;
+                while let Some(keyword) = self.parse_one_of_keywords(&[SCAN]) {
```

Comment:
What do you think about requiring comma separation here so that it's

```
OBJECTS FROM SCAN, SQS QUEUE 'blah', VIEW v
```

rather than e.g.

```
OBJECTS FROM SCAN SQS QUEUE 'blah' VIEW v
```

where I feel like the `SCAN` and `SQS` tokens run together in an unfortunate way.

Reply:
oh yeah that is 100% a bug on my part.